src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -188,18 +201,38 @@ public boolean updateResourceState(
   }
 
   /**
-   * Modifies the {@link CleanupFlightState} of a single flight. Returns whether there was a flight
-   * with the flightId.
+   * Modifies the {@link CleanupFlightState} of a single flight. Returns the updated CleanupFlight,
+   * if one was updated.
    */
   @Transactional(propagation = Propagation.SUPPORTS)
-  public boolean updateFlightState(String flightId, CleanupFlightState flightState) {
+  public Optional<CleanupFlight> updateFlightState(
+      String flightId, CleanupFlightState flightState) {
     String sql =
-        "UPDATE cleanup_flight SET flight_state = :flight_state WHERE flight_id = :flight_id;";
+        "UPDATE cleanup_flight SET flight_state = :flight_state WHERE flight_id = :flight_id "
+            + "RETURNING flight_id, flight_state";
     MapSqlParameterSource params =
         new MapSqlParameterSource()
             .addValue("flight_state", flightState.toString())
             .addValue("flight_id", flightId);
-    return jdbcTemplate.update(sql, params) == 1;
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(jdbcTemplate.query(sql, params, CLEANUP_FLIGHT_ROW_MAPPER)));
+  }
+
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public Map<String, String> retrieveLabels(TrackedResourceId trackedResourceId) {
```

Comment:
Is this needed by flight or just something we will need in the future?

Reply:
Oops, this was already included in the previous PR #29, just hadn't done the right merging yet.